### PR TITLE
Replace instances of useContractReader() with unique hooks

### PR DESCRIPTION
**Pull request by peripheralist**
## What does this PR do and why?

Wraps most contractReader instances in unique hooks. This allows for
- Better code organization
- Typed arguments to contract reads
- Less code duplication

This is still a draft for now while some uncertainties are sorted out:
- How much should reader hooks depend on Contexts? This is tricky as many of these hooks are used to define context values
- Should passed arguments be typed as an object?

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

File: src/components/Dashboard/ReservedTokens.tsx [hunk 1]
```diff
@@ -124,7 +74,7 @@ export default function ReservedTokens({
           }}
         >
           <span>
-            {formatWad(reservedTickets, { precision: 0 }) || 0}{' '}
+            {formatWad(reservedTokens, { precision: 0 }) || 0}{' '}
```
Comment (aeolianeth) [inline src/components/Dashboard/ReservedTokens.tsx:77]: <3

File: src/components/Dashboard/Rewards.tsx [hunk 2]
```diff
@@ -217,9 +157,9 @@ export default function Rewards({
                     <div>
                       {ticketsIssued && (
                         <div>
-                          {ticketsBalance?.gt(0) ? (
+                          {claimedBalance?.gt(0) ? (
```
Comment (aeolianeth) [inline src/components/Dashboard/Rewards.tsx:160]: more <3

File: src/components/Dashboard/index.tsx [hunk 3]
```diff
@@ -1,248 +1,81 @@
-import { BigNumber } from '@ethersproject/bignumber'
 import { Trans } from '@lingui/macro'
 import FeedbackFormLink from 'components/shared/FeedbackFormLink'
 
 import { ProjectContext, ProjectContextType } from 'contexts/projectContext'
-import { utils } from 'ethers'
-import useContractReader from 'hooks/ContractReader'
+import useBalanceOfProject from 'hooks/contractReader/BalanceOfProject'
```
Comment (aeolianeth) [inline src/components/Dashboard/index.tsx:5]: oh so sexy. we can basically have V2 versions of all of these, possibly

**Review by aeolianeth — APPROVED** [review]
Thanks so much for this PR @peripheralist. That `Dashboard/index.tsx` diff is just beautiful ✨ .

Admittedly I didn't check whether each of these new hook files matched the previous useContractReader calls, but I clicked around the interface and didn't have any problems.

Approving on my end 🚀

**Review by johnnyd-eth — APPROVED** [review]
Such a great tidy-up Peri, awesome work. Can't see any problems on my end, all looks good to me ✅

**Pull request merged**